### PR TITLE
Support PR review-comment creation in Prow GHC

### PR DESCRIPTION
**Pull request by rahulbharuka**
Implements #20567

### CODE REVIEW

**Review by petr-muller — APPROVED** [review]
/ok-to-test

File: prow/github/types.go [hunk 1]
```diff
@@ -912,7 +923,11 @@ type ReviewComment struct {
 	UpdatedAt time.Time `json:"updated_at"`
 	// Position will be nil if the code has changed such that the comment is no
 	// longer relevant.
-	Position *int `json:"position"`
+	Position  *int     `json:"position"`
```
Comment (fejta) [inline prow/github/types.go:926]: Can we omitempty these when they are unset?

**Review by fejta — APPROVED** [review]
/hold

File: prow/github/types.go (hunk 1 above)
Comment (rahulbharuka) [reply prow/github/types.go]: Fixed it. Thanks.

**Review by fejta — APPROVED** [review]
/hold cancel

**Pull request merged**